examples/src/bin/gio_task.rs

```diff
@@ -0,0 +1,188 @@
+//! # GTask Sample
+//!
+//! This sample demonstrates how to implement a GIO async/finish API based on GTask (see
+//! https://developer.gnome.org/gio/stable/GTask.html) in Rust using the GTask generated bindings.
+//! This can be useful, for example, when porting to Rust some existing C code exposing such an API.
+
+use glib::subclass;
+use glib::subclass::prelude::*;
+use glib::translate::ToGlibPtr;
+use gtk::glib::translate::FromGlibPtrNone;
+use gtk::prelude::AsyncResultExt;
+use gtk::prelude::Cast;
+use gtk::prelude::FileExt;
+use gtk::{gio, glib};
+use std::convert::TryInto;
+
+pub mod imp {
+    use super::*;
+
+    // FileSize is a simple object that will just contain the read file size.
+    // Initially the size field will be initialized to 0.
+    pub type FileSize = <FileSizePrivate as ObjectSubclass>::Instance;
+    pub type FileSizeClass = <FileSizePrivate as ObjectSubclass>::Class;
+    pub struct FileSizePrivate {
```

Comment:
```suggestion
    pub struct FileSize {
```

We have namespaces in Rust :)